debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/memory/MemoryLogMinerEventProcessor.java

```diff
@@ -85,221 +50,28 @@ protected MemoryTransaction createTransaction(LogMinerEventRow row) {
-            // watermark is advanced after a long transaction is committed.
-            LOGGER.warn("Event for transaction {} has already been processed, skipped.", transactionId);
-        }
+    public LogMinerCache<String, MemoryTransaction> getTransactionCache() {
+        return transactionCache;
     }
 
     @Override
-    protected int getTransactionEventCount(MemoryTransaction transaction) {
-        return transaction.getEvents().size();
-    }
+    public LogMinerCache<String, LogMinerEvent> getEventCache() {
 
-    @Override
-    protected PreparedStatement createQueryStatement() throws SQLException {
-        return jdbcConnection.connection().prepareStatement(getQueryString(),
-                ResultSet.TYPE_FORWARD_ONLY,
-                ResultSet.CONCUR_READ_ONLY,
-                ResultSet.HOLD_CURSORS_OVER_COMMIT);
+        return eventCache;
     }
 
     @Override
-    protected Scn calculateNewStartScn(Scn endScn, Scn maxCommittedScn) throws InterruptedException {
-        if (getConfig().isLobEnabled()) {
-            if (transactionCache.isEmpty() && !maxCommittedScn.isNull()) {
-                offsetContext.setScn(maxCommittedScn);
-                dispatcher.dispatchHeartbeatEvent(partition, offsetContext);
-            }
-            else {
-                abandonTransactions(getConfig().getLogMiningTransactionRetention());
-                final Scn minStartScn = getTransactionCacheMinimumScn();
-                if (!minStartScn.isNull()) {
-                    recentlyProcessedTransactionsCache.entrySet().removeIf(entry -> entry.getValue().compareTo(minStartScn) < 0);
-                    schemaChangesCache.removeIf(scn -> scn.compareTo(minStartScn) < 0);
-                    offsetContext.setScn(minStartScn.subtract(Scn.valueOf(1)));
-                    dispatcher.dispatchHeartbeatEvent(partition, offsetContext);
-                }
-            }
-            return offsetContext.getScn();
-        }
-        else {
-            if (!getLastProcessedScn().isNull() && getLastProcessedScn().compareTo(endScn) < 0) {
-                // If the last processed SCN is before the endScn we need to use the last processed SCN as the
-                // next starting point as the LGWR buffer didn't flush all entries from memory to disk yet.
-                endScn = getLastProcessedScn();
-            }
-
-            if (transactionCache.isEmpty()) {
-                offsetContext.setScn(endScn);
-                dispatcher.dispatchHeartbeatEvent(partition, offsetContext);
-            }
-            else {
-                abandonTransactions(getConfig().getLogMiningTransactionRetention());
-                final Scn minStartScn = getTransactionCacheMinimumScn();
-                if (!minStartScn.isNull()) {
-                    offsetContext.setScn(minStartScn.subtract(Scn.valueOf(1)));
-                    dispatcher.dispatchHeartbeatEvent(partition, offsetContext);
-                }
-            }
```

Comment:
Found the issue, this part is missing in the calculation, which causes the test to fail.  Reviewing the code more clearly but at least I know the culprit.